### PR TITLE
Assign a clearer notion of ownership on the TaskExec pointers

### PR DESCRIPTION
**Pull request by eliemichel**
Unless specified, the Render object owns its TaskExecs contained in m_tasks.

All methods modifying this ownership by either giving it up or taking over a
new task are commented in their docstring.

There should now be no memory leak of TaskExecs any more (not related to this at least).

Implement the point 2) given in [this comment](https://github.com/CGRU/cgru/commit/b377fc3b2ae981920ecb812ef181a2ef104d3325#commitcomment-17643608)

@timurhai This is an alternative to 99e38ca. Contrary to [what you feared](https://github.com/CGRU/cgru/commit/eb581d37aa240cbffaba039702ea497f958e900f#commitcomment-17644603), this does not introduce much more code. And also note that it does not make any extra memory allocation. If you like it, I suggest that we remove the `bool` return value you just added since it should not be needed any more. Unless you have other ideas of use for it in the furure.

### CODE REVIEW

File: afanasy/src/server/block.cpp [hunk 1]
```diff
@@ -208,10 +208,10 @@ bool Block::v_startTask( af::TaskExec * taskexec, RenderAf * render, MonitorCont
    return true;
 }
 
-void Block::reconnectTask( af::TaskExec & i_taskexec, RenderAf & i_render, MonitorContainer * i_monitoring)
+void Block::reconnectTask(af::TaskExec *i_taskexec, RenderAf & i_render, MonitorContainer * i_monitoring)
 {
-	Task * task = m_tasks[i_taskexec.getTaskNum()];
-	task->reconnect( &i_taskexec, m_data->getRunningTasksCounter(), &i_render, i_monitoring);
```
Comment (eliemichel) [inline afanasy/src/server/block.cpp]: Note that I switched back the references to pointers here in arguments.
I prefer references in general, which is why I used it at first, but it does not make sense to "own" a reference as it makes for pointers, so it is cleaner imho to use pointers when there is a notion of owning transmission.

**Comment by timurhai** [comment]
Yes, afwatch can easily take ownership.
And on normal server work, RenderAf adds and then removes taskexecs from array.
But on server exit and RenderAf and TaskRun will own taskexec.

**Comment by eliemichel** [comment]
Arf I did not followed my changes all the way, will fix this.

**Comment by timurhai** [comment]
I have some idea for feature.
af::Render and RenderAf can have own taskexecs array.
af::Render shoudl always free its own array.
RenderAf shoudl never free it.
On reconnect we can remove successfully reconnected execs from af::Render.
On server side on renders list request, RenderAf can create temp array in af::Render,
like it performed here:
https://github.com/CGRU/cgru/blob/master/afanasy/src/render/renderhost.cpp#L209
Or:
We can make a virtual function rw_tasksexecs, this way, af::Render and RenderAf can write own arrays to the message.

**Comment by eliemichel** [comment]
Right at the same time I committed a small change making the `TaskExec` _never_ owned by `TaskRun`, except when it removes it, but then it directly deletes it.

**Comment by timurhai** [comment]
Yes.
`af::TaskExec` should own `af::Render` as it designed to send data to it.
But how we should merge this branch now?
Just make a new commit (by hand) in master that will make the same?

**Comment by eliemichel** [comment]
I'll merge it manually, yes, if you agree w/ it

**Comment by eliemichel** [comment]
Merged by 53b1766

**Pull request closed without merging**